Attack vectors blog post

kevgo opened the pull request:
resolves https://github.com/ory-corp/general/issues/546

replaces #661 

I have fixed the `attack_vectors` branch and copied all my edits from my fork into this file. 


## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [x] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added necessary documentation within the code base (if
      appropriate).

kevgo commented on src/markdown/blog/attack_vectors.mdx:
```diff
@@ -0,0 +1,169 @@
+acc.google.com and dev.google.com. dev.google.com redirects to acc.google.com.
+After some testing with redirection, Andrew can perform an attack.
+
+He sends George a simple message like, "Hey, look what Google introduced in the
```
In the paragraph above, you set up `acc.google.com` and `dev.google.com`. Here you mention `dev.account.com`. I don't see the connection.

kevgo commented:
@gen1us2k I have copied the comments from the original PR over. Please take a look.

gen1us2k commented on src/markdown/blog/attack_vectors.mdx:
```diff
@@ -0,0 +1,169 @@
+the link, gets redirected to acc.google.com, enters credentials, and sees a
+webpage. Alas, his account credentials was compromised a couple of minutes ago
+when he entered his credentials on a phishing page. Andrew needs to take his
+phone for a couple of minutes to gain full access to George's Google account.
```
I added additional context to the story.

kevgo commented on .prettierrc.js:
```diff
@@ -1,12 +1,20 @@
-   "^\\.(?!.*module\\.css$)(.*)\\.css$",
-   "^\\.(.*)\\.module\\.css$",
-    "^\\.(.*)\\.(?!css$)[a-z]+$",
+  importOrder: [
```
This is just temporary to keep the printWidth at manageable levels until https://github.com/ory/prettier-styles/pull/42 ships. I'll revert the changes in this file before shipping.

kevgo commented on src/markdown/blog/attack_vectors.mdx:
This example is way more detailed than the other ones. I suggest sticking to the same detail level for all examples. Is it okay if we leave out this example here?

vinckr replied:
As explained below I think the example should either be simplified or if that is not great you can leave it out.

kevgo commented on src/markdown/blog/attack_vectors.mdx:
This example requires something very difficult (getting access to the unlocked phone of the victim) to perform a simple attack (get a logged in session once). As such, it leaves a lot of questions open.

Andrew could have achieved the same with a much simpler man-in-the-middle attack since he got George to open the link and log in already. In this situation, George would have no issue clicking on the 2FA popup on his own. No need for Andrew to borrow the phone!

vinckr replied:
I agree the example is good, but it seems like super complex to actually do the attack.
If the person trusts me as much to get unsupervised access to their phone (IMO today often even best friends or very close people wouldn't give out access to their phones like that). 

The alternative "story" without the phone borrowing would be better or rather a more striking/relatable example.

kevgo reviewed (commented):
I have finished another round of polish. Some suggestions below.

kevgo commented:
Okay, I have simplified the example. I have also re-arranged the content a bit and split one of the chapters into two. I have also added some diagrams using https://excalidraw.com.

I think in general informative blog posts like this one make the most sense if they contain this information:

1. Here is an interesting problem and why it's important for you to know about it.
2. Here is some useful knowledge and best practices what to do about it.
3. Here is how the respective Ory solution implements these principles.

Those are at least the blog posts I like to read from vendors. If I just need (2), I can read Wikipedia. If I read the newsletter of a solution I use, I appreciate technical details around how they work and why that makes sense. As an example, I am subscribed to the Fedora, Gnome, Go, and Rust blogs and appreciate insights into the thinking, design principles, and decisions that go into these systems.

kevgo commented:
I have updated this branch and resolved merge conflicts after doing some large-scale code repair in this code base. You are welcome. Please run `npm i --legacy-peer-deps` to install the updated configuration and tooling. Thanks! 🤘
